### DISCUSSION TITLE
Device initialization failed for external flash MX66L1G with nRF5340 microcontroller

### DISCUSSION THREAD

**Discussion by RadhikaMaheshwari-eaton** [zephyrproject-rtos/zephyr#76763] [Q&A]
Hello,

I am working on a Zephyr example for SPI flash zephyr/samples/drivers/spi_flash using the nRF Connect SDK 2.6.1 on the nRF5340 development board. My SPI flash is a serial NOR flash MX66L1G45G by Macronix. I referred to the nRF5340_cpuapp_common.dtsi file, but I am encountering a device initialization failed error (error 19). 
I suspect the issue originates from the z_sys_init_run_level function in zephyr/kernel/init.c when it executes all the initialization functions. I am still debugging and will share more findings as I progress.
I’ve attached the zip file of the SPI flash example I’m using. If anyone can help me understand the issue, it would be greatly appreciated.
[spi_flash.zip](https://github.com/user-attachments/files/16519011/spi_flash.zip)

**Comment by RadhikaMaheshwari-eaton** [comment]
Hi,

I encountered the error message [00:00:00.253,265] <err> qspi_nor: JEDEC id [00 00 00] expect [c2 20 1b] in the qspi_init function located in zephyr\drivers\flash\nrf_qspi_nor.c
![image](https://github.com/user-attachments/assets/661bd9df-bd71-420a-9ee4-d4c526a8e3b4)
I have specified the JEDEC ID in the overlay since it is a required property for Nordic’s QSPI NOR. 
![image](https://github.com/user-attachments/assets/8fe1f08b-34a3-4591-9cf7-20ae532214bd)
This JEDEC ID is also correct, as verified from the datasheet of the external flash MX66L1G. 
![image](https://github.com/user-attachments/assets/1af9adfa-9e62-4551-9619-fe25e5d103b0)

**Reply by nordicjm** [reply]
Your device cannot be queried, check the connections, power, and that the device is in QSPI mode

**Comment by koffes** [comment]
This basically means that the QSPI receives no response from the external flash.
At this point is is often a good idea to rule out either a hardware or software issue. Do you have access to a logic analyzer? I would probe the QSPI lines and see if there is indeed a request and/or a response.

If there is no data from the 5340: Check that the QSPI pin configurations are correct and that there are no shorts to ground. Check that the solder bridges on the DK are in the correct config. Check that the debug switch is turned on.

If there is data just from the 5340 but no response: Check that the CS pin asserts (goes low) as it should, measure that the ext flash has power.
